ldclient/__init__.py

```diff
@@ -43,16 +47,121 @@ def emit(self, record):
 
 class Config(object):
 
-    def __init__(self, base_uri, connect_timeout = 2, read_timeout = 10, upload_limit = 100, capacity = 10000):
+    def __init__(self,
+                 base_uri = 'https://app.launchdarkly.com',
+                 connect_timeout = 2,
+                 read_timeout = 10,
+                 upload_limit = 100,
+                 capacity = 10000,
+                 stream_uri = 'https://stream.launchdarkly.com',
+                 stream = False,
+                 verify = True):
         self._base_uri = base_uri.rstrip('\\')
+        self._stream_uri = stream_uri.rstrip('\\')
+        self._stream = stream
         self._connect = connect_timeout
         self._read = read_timeout
         self._upload_limit = upload_limit
         self._capacity = capacity
+        self._verify = verify
 
     @classmethod
     def default(cls):
-        return cls('https://app.launchdarkly.com')
+        return cls()
+
+class InMemoryFeatureStore(object):
+    def __init__(self):
+        self._lock = ReadWriteLock()
+        self._initialized = False
+        self._features = {}
+
+    def get(self, key):
+        try:
+            self._lock.rlock()
+            f = self._features[key]
+            if f is None or 'deleted' in f and f['deleted']:
+                return None
+            return f
+        finally:
+            self._lock.runlock()
+
+    def all(self):
+        try:
+            self._lock.rlock()
+            return dict((k,f) for k,f in self._features.iteritems() if ('deleted' not in f) or not f['deleted'])
+        finally:
+            self._lock.runlock()
+
+    def init(self, features):
+        try:
+            self._lock.lock()
+            self._features = dict(features)
+            self._initialized = True
+        finally:
+            self._lock.unlock()
+
+    def delete(self, key, version):
+        try: 
+            self._lock.lock()
+            f = self._features[key]
+            if f is not None and f['version'] < version:
+                f['deleted'] = True
+                f['version'] = version
+            elif f is None:
+                f = {'deleted': True, 'version': version}
+                self._features[key] = f
+        finally:
+            self._lock.unlock()        
+
+    def upsert(self, key, feature):
+        try:
+            self._lock.lock()
+            f = self._features[key]
```

Comment:
I think this will throw a `KeyError` if the feature is not there (instead of `None`, which we are checking for on the next line).

    f = self._features.get(key, None)

I think should do what we want.